File: modules/facade/src/test/java/com/teamwizardry/librarianlib/facade/example/containers/DirtSetterItem.java
package com.teamwizardry.librarianlib.facade.example.containers;

import net.minecraft.item.Item;
import net.minecraft.item.ItemUsageContext;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.TranslatableText;
import net.minecraft.util.ActionResult;

import com.teamwizardry.librarianlib.facade.example.ExampleModContainers;
import org.jetbrains.annotations.NotNull;

public class DirtSetterItem extends Item {
    public DirtSetterItem(@NotNull Item.Settings properties) {
        super(properties);
    }

    @Override
    public ActionResult useOnBlock(ItemUsageContext context) {
        if (!context.getWorld().isClient) {
            ExampleModContainers.dirtSetterContainerType.open(
                    (ServerPlayerEntity) context.getPlayer(),
                    new TranslatableText("modid.container.dirt_setter"),
                    // additional constructor arguments:
                    context.getBlockPos()
            );
        }
        return ActionResult.SUCCESS;
    }
}

File: modules/foundation/src/test/kotlin/com/teamwizardry/librarianlib/foundation/testmod/customtypes/DirtSetterItem.kt
package com.teamwizardry.librarianlib.foundation.testmod.customtypes

import com.teamwizardry.librarianlib.foundation.item.BaseItem
import com.teamwizardry.librarianlib.foundation.testmod.ModContainers
import net.minecraft.item.ItemUseContext
import net.minecraft.util.ActionResultType
import net.minecraft.entity.player.ServerPlayerEntity
import net.minecraft.util.text.TranslationTextComponent

class DirtSetterItem(properties: Properties): BaseItem(properties) {
    override fun onItemUse(context: ItemUseContext): ActionResultType {
        if (!context.world.isRemote) {
            ModContainers.dirtSetter.open(
                (context.player as ServerPlayerEntity?)!!,
                TranslationTextComponent("librarianlib-foundation.test_container.title"),
                context.pos
            )
        }
        return ActionResultType.SUCCESS
    }
}